### DISCUSSION TITLE
Hook for detecting if a route is currently matched

### DISCUSSION THREAD

**Discussion by ryanstaniforth** [TanStack/router#376] [Ideas]
I have a use case for detecting if a child route is matched. I'm currently using the following:

```typescript
const matches = useMatches();
const isUserSelected = matches.find((match) => match.routeId === '/authenticated/users/:id') !== undefined;
```

I think something like this would be ideal, _including typings for the available routes_:

```typescript
const isUserSelected = useIsMatched('/authenticated/users/:id');
```

This functionality is similar to how `activeProps` on a link `<Link>` works. 

P.S. Thank you for this amazing library @tannerlinsley, loving it so far.

**Comment by PointSingularity** [comment]
@ryanstaniforth Did you find a way to do this?

I am having the same problem, trying to do some logic in the parent component, but not being able to get the current matched child route 

Edit: 
```tsx
 const matches = useMatches();
 console.log(matches.map((m) => m.id));
```
I can get them with this, but IDK if there is a better way.

**Reply by ryanstaniforth** [reply]
I'm still using `useMatches`.

**Reply by sweethuman** [reply]
Still using this

```js
const matches = useMatches();
const match = useMemo(() => matches.some(elem => elem.id === '/login'), [matches]);
```
